### DISCUSSION TITLE
Not storing refresh token in database

### DISCUSSION THREAD

**Discussion by Fersubair40** [maxgalbu/adonis5-jwt#7] [Q&A]
Hi there, thanks for this awesome package but I noticed refresh tokens get saved in the database is there a way to bypass that not to store the refresh token in the database

**Comment by maxgalbu** [comment]
Hi @Fersubair40, where would you store it?

You could use a JWT for refresh token, but then you completely lose the ability to
- revoke refresh tokens quickly 
- prevent an user from logging in automatically
- allow compromised refresh tokens to be revoked

**Reply by Fersubair40** [reply]
Thanks, @maxgalbu for your response JWT is the primary mode of authentication so storing the refresh token once JWT is generated might lead to unnecessary memory usage  on the database and the frontend application doesn't really use the refresh token

so is there a way to use JWT for refresh token?


Alternatively could probably use Redis for storage since that would clear the refresh token after expiration but the budget resources might be small to run a Redis instance.

Thanks looking forward to your reply.

**Reply by maxgalbu** [reply]
> for your response JWT is the primary mode of authentication so storing the refresh token once JWT is generated might lead to unnecessary memory usage on the database and the frontend application doesn't really use the refresh token

I guess you can say "the frontend application doesn't really use the refresh token" if you set the JWT expiration date to 1 year 😄 which is not recommended.

This opens a whole lot of issues if someone steals your users' JWT, and you have no way to revoke the (still valid) JWT. And even with a short-lived JWT and a JWT refresh token, you still have no way to revoke access to that user, because he holds both the JWT and the refresh token.

So, you could shoot yourself in the foot with this solution 😁 is this what you want?
